buildSrc/src/main/kotlin/CodeCoverage.kt

```diff
@@ -0,0 +1,209 @@
+import com.android.build.gradle.BaseExtension
+import com.android.build.gradle.api.BaseVariant
+import org.gradle.api.Project
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.tasks.testing.Test
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.register
+import org.gradle.kotlin.dsl.withType
+import org.gradle.testing.jacoco.plugins.JacocoPluginExtension
+import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
+import org.gradle.testing.jacoco.tasks.JacocoCoverageVerification
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+object CodeCoverage {
+    const val JACOCO_VERSION = "0.8.3"
+
+    val generatedClasses = setOf(
+            "**/R.class",
+            "**/R$*.class",
+            "**/BuildConfig.*",
+            "**/Manifest*.*",
+            "com/android/**/*.class",
+            "**/*_ViewBind*" // ButterKnife auto generated classes
+    )
+
+    val androidClasses = setOf(
+            "**/*Activity.class",
```

Comment:
I thought our approach was going to be that we wanted a certain % of code coverage across all the code, not excluding view code.

Reply:
With the coverageMinimum tasks being added, the expectation is that code coverage will be enforced. Since Android classes are hard/impossible to test, there would be a high variance on the coverage depending on what the story/feature asks for. This would cause PRs that deals mainly with Android classes to decrease the coverage and fail the build, so they would either have to temporarily lower the minimum threshold or go out of way to test those Android classes to the point that it end up being counterproductive.
This is also expecting that there are also UI tests in addition to unit tests. And those tests would include Android classes in the coverage and have a separate minimum threshold.

It's pretty easy to change the excluded classes though, the caller would just pass in a different set to the `unitTestExcludedClasses` parameter. I'm open to changing the default parameter to just emptySet().